### ISSUE TITLE
Concurrent Mode compatibility? 😃 

### ISSUE THREAD

**Issue by sesam** [merisbahti/klyva#45]
In an older version of https://github.com/dai-shi/will-this-react-global-state-work-in-concurrent-rendering we had klyva included. At that time, klyva was passing all tests except this one:

    check with useTransition
      ✕ check 7: proper branching with transition (6127 ms)

What is this test 7 checking? Is this something that is hard to fix in klyva? If it needs a de-optimisation, could it be as an optional extra or a plugin or monkeypatch under a contrib/  folder, so that klyva could be the first to pass all the known tests for concurrent mode? 🎉

**Comment by merisbahti** [comment]
Yeah that's a fun question, I haven't really seen any other (feature rich) state management solution succeed here.

I think a good start would be adding klyva to dai-shi's repo again, just to see how well klyva is faring nowadays - and take it from there.

I'll see if I can do it during the weekend :D
